pkg/resources/validation/cluster/provisioningcluster.go

```diff
@@ -135,6 +147,36 @@ func validateCreatorID(request *webhook.Request, oldCluster, cluster *v1.Cluster
 	return nil
 }
 
+func (p *provisioningClusterValidator) validateClusterName(request *webhook.Request, response *webhook.Response, cluster *v1.Cluster) error {
```

Comment:
So `cluster` is coming from `request` so ideally we want to either pass just `request` and have func extract what it needs or pass individual pieces it needs from `request`. In this case, I would probably go with latter as the only piece we need from request is operation.
The rationale here is that current signature allows passing `request` that's for cluster that is different from passed `cluster`.
I see other funcs here do the same, so I'm not hell-bent on having this fixed.

Reply:
I understand where you are coming from, but I would like to keep the code consistent for now.